numpy/lib/histograms.py

```diff
@@ -97,6 +97,39 @@ def _hist_bin_scott(x):
     return (24.0 * np.pi**0.5 / x.size)**(1.0 / 3.0) * np.std(x)
 
 
+def _hist_bin_ise(x):
+    """
+    Histogram bin estimator based on minimizing the estimated integrated squared error (ISE).
+
+    The number of bins is chosen by minimizing the estimated ISE against the unknown true distribution.
+    The ISE is estimated using cross-validation and can be regarded as a generalization of Scott's rule.
+    https://en.wikipedia.org/wiki/Histogram#Scott.27s_normal_reference_rule
+
+    Parameters
+    ----------
+    x : array_like
+        Input data that is to be histogrammed, trimmed to range. May not
+        be empty.
+
+    Returns
+    -------
+    h : An estimate of the optimal bin width for the given data.
+    """
+
+    n = x.size
+    ptp_x = np.ptp(x)
+    if n <= 1 or ptp_x == 0:
+        return ptp_x
```

Comment:
Would't this be better spelt `return 0`?

Right now `x = [inf]` causes NaN to be returned, which may not be deliberate.

Reply:
This problem will be handled before this function is called.
https://github.com/numpy/numpy/blob/a10bf46fe7186e0555fbff55dcef4cd3291724a3/numpy/lib/histograms.py#L231-L233

Reply:
In which case, `return 0` would definitely be clearer here